src/main/MQ2Spawns.cpp

```diff
@@ -81,195 +72,464 @@ static bool dataNamingSpawn(const char* szIndex, MQTypeVar& Ret)
+
+		bool invis = pSpawn->HideMode != 0;
+
+		if (invis)
+			fmt::format_to(buf, "({})", pSpawn->DisplayedName);
+		else
+			fmt::format_to(buf, "{}", pSpawn->DisplayedName);
+
+		if (showNamesLevel > 1 && pSpawn->Lastname[0] != 0)
+			fmt::format_to(buf, " {}", pSpawn->Lastname);
+
+		if (showNamesLevel > 3 && pSpawn->Suffix[0] != 0)
+			fmt::format_to(buf, " {}", pSpawn->Suffix);
+
+		if (pSpawn->LFG)
+			fmt::format_to(buf, " LFG");
+		if (pSpawn->AFK != 0)
+			fmt::format_to(buf, " AFK");
+		if (pSpawn->Linkdead)
+			fmt::format_to(buf, " LD");
+
+		if (pSpawn == pTarget)
+			fmt::format_to(buf, " - {}%", static_cast<int>(pSpawn->HPMax == 0 ? 0 : pSpawn->HPCurrent * 100 / pSpawn->HPMax));
+		
+		if (showNamesLevel == 3 || showNamesLevel == 4)
+		{
+			if (pSpawn->GuildID != -1)
+			{
+				const char* szGuild = GetGuildByID(pSpawn->GuildID);
+				if (szGuild)
+				{
+					fmt::format_to(buf, "\n<{}>", szGuild);
+				}
+			}
+		}
+
+		break;
+	}
+
+	case Caption_NPC:
+	case Caption_Pet:
+	case Caption_Merc: {
+		int mark = GetNPCMarkNumber(pSpawn);
+		if (mark != 0)
+			fmt::format_to(buf, "{} - ", mark);
+
+		bool assist = IsAssistNPC(pSpawn);
+
+		if (assist)
+			fmt::format_to(buf, " >> ");
+
+		fmt::format_to(buf, "{}", pSpawn->DisplayedName);
+
+		if (pSpawn == pTarget || assist || mark != 0)
+			fmt::format_to(buf, " - {}%", static_cast<int>(pSpawn->HPMax == 0 ? 0 : pSpawn->HPCurrent * 100 / pSpawn->HPMax));
+
+		if (assist)
+			fmt::format_to(buf, " <<");
+
+		if (pSpawn->Lastname[0] != 0)
+		{
+			fmt::format_to(buf, "\n({})", pSpawn->Lastname);
+		}
+		break;
 	}
-	else if (!_stricmp(Arg1, "Player5"))
+
+	case Caption_Corpse:
+		fmt::format_to(buf, "{}'s corpse", pSpawn->DisplayedName);
+		break;
+
+	default:
+		return {};
+	};
+
+	return fmt::to_string(buffer);
+}
+
+enum eCaptionColor
+{
+	CC_PC,
+	CC_PCConColor,
+	CC_PCRaidColor,
+	CC_PCClassColor,
+	CC_PCGroupColor,
+	CC_PCTrader,
+	CC_NPC,
+	CC_NPCConColor,
+	CC_NPCClassColor,
+	CC_NPCMerchant,
+	CC_NPCBanker,
+	CC_NPCAssist,
+	CC_NPCMark,
+	CC_PetNPC,
+	CC_PetPC,
+	CC_PetConColor,
+	CC_PetClassColor,
+	CC_Mercenary,
+	CC_Corpse,
+	CC_CorpseClassColor,
+};
+
+struct MQCaptionColor
+{
+	const char*   szName;
+	const char*   szDescription;
+	bool          Enabled;
+	bool          DefaultEnabled;
+	bool          ToggleOnly;
+	const MQColor DefaultColor;
+	MQColor       Color;
+
+	MQCaptionColor(const char* name, const char* description, bool defaultEnabled, bool toggleOnly, MQColor defaultColor)
+		: szName(name), szDescription(description), DefaultEnabled(defaultEnabled), Enabled(defaultEnabled)
+		, ToggleOnly(toggleOnly), DefaultColor(defaultColor), Color(defaultColor)
 	{
-		pCaption = gszSpawnPlayerName[5];
 	}
-	else if (!_stricmp(Arg1, "Player6"))
+
+	void Save()
 	{
-		pCaption = gszSpawnPlayerName[6];
+		WritePrivateProfileString("Caption Colors", szName, Enabled ? "ON" : "OFF", mq::internal_paths::MQini);
+
+		if (!ToggleOnly)
+		{
+			char key[24], value[64];
```

Comment:
Consider replacing the magic numbers here with constexpr to show what they represent.